packages/optimizer/spec/lib/HtmlDomHelper.test.js

```diff
@@ -28,6 +28,21 @@ test('findMetaViewport returns null if tag is not present', async () => {
   expect(result).toBeNull();
 });
 
+test('works even when template is the final node', async () => {
+  let err = null;
+
+  try {
+    await treeParser.parse(`<!doctype html>
```

Comment:
Does this test ever fail?

Reply:
it would fail if the code I added isn't there

Reply:
ok, interesting. skipNodeAndChildren is only called from within transformers and these shouldn't be run when calling treeParser.parse.